src/deviceOauthService.ts

```diff
@@ -175,21 +189,21 @@ export class DeviceOauthService extends AsyncCreatable<OAuth2Config> {
     );
     try {
       return await makeRequest<DeviceCodePollingResponse>(httpRequest);
-    } catch (e) {
-      /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/restrict-template-expressions */
-      const err: any = (e as SfError).data;
-      if (err.error && err.status === 400 && err.error === 'authorization_pending') {
+    } catch (e: unknown) {
+      const err = (
+        e instanceof SfError ? e.data : SfError.wrap(isString(e) ? e : 'unknown').data
```

Comment:
checking the type `as`sertion logic:
in the case of string error, 
  .data would be `undefined`
  so then err would be `undefined`

Then the type for `err` should be `DeviceCodeAuthError | undefined`.

[you've got all the `?` in place to handle that properly, I'm thinking of the next person who has to touch this and TS lets them refer to `err` without them realizing that it *could* be undefined and then the dishonest `as`sertion leads to a runtime TypeError ]